### PR TITLE
network: make MACAddress= can take longer address

### PR DESCRIPTION
**Pull request by yuwata**
Split out of #21255.

### CODE REVIEW

File: src/network/networkd-setlink.c [hunk 1]
```diff
@@ -183,7 +184,7 @@ static int link_set_mac_allow_retry_handler(sd_netlink *rtnl, sd_netlink_message
                 return 0;
         }
 
-        /* set_link_mac_handler() also decrement set_link_messages, so once increment the value. */
+        /* set_link_mac_handler() also decrements set_link_messages, so once increments the value. */
```
Comment (bluca) [inline src/network/networkd-setlink.c]: "so increment the value once"

**Review by bluca — APPROVED** [review]
lgtm

**Comment by yuwata** [comment]
@bluca Thank you for your review. Upgrading the green label.

File: src/shared/netif-util.c [hunk 2]
```diff
@@ -99,3 +103,91 @@ int net_get_unique_predictable_data_from_name(
...
+                if (ether_addr_is_multicast(&new_hw_addr->ether)) {
+                        if (warn_invalid)
+                                log_link_warning(&link, "Specified MAC address has multicast bit set, clearing the bit.");
+
+                        new_hw_addr->bytes[0] &= 0xfe;
+                }
+
+                if (!ether_addr_is_local(&new_hw_addr->ether)) {
+                        if (warn_invalid)
+                                log_link_warning(&link, "Specified MAC address has not local assignment bit set, setting the bit.");
```
Comment (keszybz) [inline src/shared/netif-util.c:158]: has not → doesn't have the

File: src/network/networkd-setlink.c [hunk 3]
```diff
@@ -542,7 +543,7 @@ static bool link_is_ready_to_call_set_link(Request *req) {
                 break;
         case SET_LINK_MAC:
                 if (req->netlink_handler == link_set_mac_handler) {
-                        /* This is the second trial to set MTU. On the first attempt
+                        /* This is the second trial to set hardware address. On the first attempt
```
Comment (keszybz) [inline src/network/networkd-setlink.c:546]: "trial" should be either "try" or "attempt".

**Review by keszybz — CHANGES_REQUESTED** [review]
LGTM.

Please add the cosmetic adjustments to some later PR.

**Pull request merged**